.circleci/config.yml

```diff
@@ -91,6 +91,17 @@ jobs:
       - run: stack exec regression
       - run: stack exec gaussian_process
       - run: stack exec vae
+  nix-build:
+    docker:
+      - image: "nixos/nix:latest"
+    steps:
+      - run: nix-env -i git
+      - checkout
+      - run: git submodule init && git submodule update
+      - run: nix-env -iA cachix -f https://cachix.org/api/v1/install
+      - run: cachix use hasktorch
+      - run: cd deps;nix-shell --pure -p python37.pkgs.pyyaml --run "./get-deps.sh -s"
+      - run: nix-build
```

Comment:
tbh I'm not sure what will happen here, because there are several derivations in the `default.nix`.
In that case, say
```
nix-build -A hasktorch
```

Reply:
I want to tests all packages of this repo like codegen and examples.
Does the command of `nix-build -A hasktorch` test the packages?

Reply:
ah, so you want to also build the `hasktorch-codegen` and `hasktorch-examples` derivations then. ok, just say `nix-build`. I tried it just now, and it builds all of these :)